packages/affiliates/test/mocha/affiliates.js

```diff
@@ -21,129 +32,249 @@ const devRewardsToDistribute = "50000";
-  let affiliates;
-  before(function() {
-    const queries = Queries(datasetPath);
-    const coingecko = Coingecko(datasetPath);
-    const synthPrices = SynthPrices(datasetPath);
-    affiliates = DeployerRewards({
-      queries,
-      empAbi,
-      empCreatorAbi,
-      coingecko,
-      synthPrices
+  describe("CalculateRewards Simple Data", function() {
+    let balanceHistories, params, totalRewards, affiliates;
+    before(function() {
+      const queries = Queries(datasetPath);
+      const coingecko = Coingecko(datasetPath);
+      const synthPrices = SynthPrices(datasetPath);
+      affiliates = DeployerRewards({
+        queries,
+        empAbi: empAbi,
+        empCreatorAbi: empCreatorAbi,
+        coingecko,
+        synthPrices
+      });
+
+      function makePricesWithValue(count) {
+        return [makePrices(count), affiliates.utils.calculateValue];
+      }
+      function makePrices(count) {
+        return Prices(
+          lodash.times(count, i => {
+            // [timestamp, price]: we ensure price here is not 0 so that calculations come out whole
+            return [i, toWei((i + 1).toString()).toString()];
+          })
+        );
+      }
+      const startTime = 0;
+      const endTime = 10;
+      const empWhitelist = ["a", "b", "c", "d", "e", "f"];
+      balanceHistories = empWhitelist.map(x => [x, EmpBalancesHistory()]);
+      const empDeployers = empWhitelist.map(x => [x, { deployer: x + x }]);
+      totalRewards = "100";
+
+      params = {
+        startTime,
+        endTime,
+        empWhitelist,
+        empCreatorAddress: "creator",
+        snapshotSteps: 1,
+        collateralTokenPrices: empWhitelist.map(() => makePrices(endTime - startTime)),
+        collateralTokenDecimals: empWhitelist.map(() => 18),
+        syntheticTokenPricesWithValueCalculation: empWhitelist.map(() => makePricesWithValue(endTime - startTime)),
+        syntheticTokenDecimals: empWhitelist.map(() => 18),
+        blocks: lodash.times(endTime - startTime, i => ({ timestamp: i, number: i })),
+        balanceHistories,
+        empDeployers,
+        totalRewards
+      };
     });
-  });
-  it("getBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getBalanceHistory(empContracts[0], startingTimestamp, endingTimestamp);
-    assert.ok(result);
-    assert.ok(result.history.length());
-  });
-  it("getAllBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getAllBalanceHistories(empContracts, startingTimestamp, endingTimestamp);
-    assert.equal(result.length, empContracts.length);
-    result.forEach(([address, history]) => {
-      assert.ok(address);
-      assert.ok(history);
-      assert.ok(history.history.length());
+    it("should work with basic test cases", function() {
+      // add some balance histories to the emp contracts. this is really what ends up adjusting the distribution.
```

Comment:
i should probably name these tests better but here @nicholaspai 

Below we add a single position to the balance history at block 0. The datastructure passed in is what a the decoded `PositionCreated` event looks like in the emp.